### PR TITLE
Task-11: Implement an endpoint of type POST

### PR DESCRIPTION
**Pull request by cor3man**

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,3 +1,5 @@
 APP_PORT=4000
+NODE_TLS_REJECT_UNAUTHORIZED = '0'
 NASA_API_KEY=your_nasa_token
 NASA_API_NEO_URL=https://api.nasa.gov/neo/rest/v1/feed
+NASA_API_ROVER_URL=https://api.nasa.gov/mars-photos/api/v1/rovers/curiosity/latest_photos
```
Comment (mdziarkach) [inline .env.example]: let's leave only this part NASA_API_ROVER_URL=https://api.nasa.gov

File: src/controllers/roverController.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+const { getRoverImage } = require('../useCases/roverUseCase');
+
+const getRoverImageForUser = async (req, res) => {
+    try {
+        const { userId, userName, userApiKey } = req.body;
+
+        if (!userId || !userName || !userApiKey) {
+            return res.status(400).json({ error: 'Missing required user data' });
```
Comment (mdziarkach) [inline src/controllers/roverController.js]: let's create custom Exception with status code and error message
If any difficulties arise, you can always look at how other guys wrote it :)

File: src/controllers/roverController.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+            return res.render('rover.njk', { roverImage });
+        } else {
+            return res.json({
+                image: roverImage
+            });
+        }
+
+    } catch (error) {
+        console.error('Error fetching rover image:', error.message);
+        res.status(500).json({ error: 'Failed to fetch rover image' });
```
Comment (mdziarkach) [inline src/controllers/roverController.js]: `next(error)` should be enough

File: src/index.js [hunk 4]
```diff
@@ -1,22 +1,26 @@
 const express = require('express');
 const { getMeteors } = require('./controllers/meteorsController');
+const { getRoverImageForUser } = require('./controllers/roverController');
 const nunjucks = require('nunjucks');
 require('dotenv').config({ path: '../.env' });
 
 const NASA_API_KEY = process.env.NASA_API_KEY;
 const NASA_API_NEO_URL = process.env.NASA_API_NEO_URL;
+const NASA_API_ROVER_URL = process.env.NASA_API_ROVER_URL
 const PORT = process.env.APP_PORT;
```
Comment (mdziarkach) [inline src/index.js]: create please config file with all env variables

File: src/repositories/roverRepository.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    if (photos.length === 0) {
+      throw new Error('No photos available.');
+    }
+
+    const latestPhoto = photos[0];
+    return latestPhoto.img_src;
+  } catch (error) {
+    console.error('Error fetching data from NASA Rover API:', error.message);
+    throw error;
+  }
```
Comment (mdziarkach) [inline src/repositories/roverRepository.js:16]: we don't need try..catch here. We will catch all error in the error handler

File: src/repositories/roverRepository.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+    const response = await axios.get(process.env.NASA_API_ROVER_URL, {
+      params: {
+        api_key: apiKey
+      }
+    });
+
+    const photos = response.data.latest_photos;
+    if (photos.length === 0) {
+      throw new Error('No photos available.');
```
Comment (mdziarkach) [inline src/repositories/roverRepository.js]: throw please custom exception here

File: .env.example (hunk 1 above)
Comment (cor3man) [reply .env.example]: @Mikhail1992 done

File: src/controllers/roverController.js (hunk 2 above)
Comment (cor3man) [reply src/controllers/roverController.js]: @Mikhail1992 done

File: src/controllers/roverController.js (hunk 3 above)
Comment (cor3man) [reply src/controllers/roverController.js]: @Mikhail1992 done

File: src/index.js (hunk 4 above)
Comment (cor3man) [reply src/index.js]: @Mikhail1992 done

File: src/repositories/roverRepository.js (hunk 6 above)
Comment (cor3man) [reply src/repositories/roverRepository.js]: @Mikhail1992 done

File: src/repositories/roverRepository.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+const getLatestRoverImage = async (apiKey) => {
+  const response = await axios.get(NASA_API_ROVER_URL, {
+    params: {
+      api_key: apiKey
+    }
+  });
+
+  const photos = response.data.latest_photos;
+  if (photos.length === 0) {
+    throw new Error('No photos available.');
```
Comment (mdziarkach) [inline src/repositories/roverRepository.js]: CustomException should be thrown here

File: src/repositories/meteorsRepository.js [hunk 8]
```diff
@@ -1,7 +1,10 @@
+const { NASA_API_URL } = require('../configurations/config');
 const axios = require('axios');
+const NASA_API_NEO_URL = NASA_API_URL + '/neo/rest/v1/feed';
+
 
 const getMeteorsData = async (startDate, endDate) => {
-    const url = `${process.env.NASA_API_NEO_URL}?start_date=${startDate}&end_date=${endDate}&api_key=${process.env.NASA_API_KEY}`;
+    const url = `${NASA_API_NEO_URL}?start_date=${startDate}&end_date=${endDate}&api_key=${process.env.NASA_API_KEY}`;
```
Comment (mdziarkach) [inline src/repositories/meteorsRepository.js]: add please config file for environment variables manipulations

File: src/repositories/roverRepository.js (hunk 7 above)
Comment (cor3man) [reply src/repositories/roverRepository.js]: @Mikhail1992 Done

File: src/repositories/meteorsRepository.js (hunk 8 above)
Comment (cor3man) [reply src/repositories/meteorsRepository.js]: @Mikhail1992 I have config.js file already, I just forgot to import process.env.NASA_API_KEY. 
Fixed

File: src/repositories/meteorsRepository.js [hunk 9]
```diff
@@ -1,7 +1,12 @@
+const { NASA_API_URL, NASA_API_KEY} = require('../configurations/config');
 const axios = require('axios');
+const NASA_API_NEO_URL = NASA_API_URL + '/neo/rest/v1/feed';
+
+const NASA_API_NEO_URL = NASA_API_URL + '/neo/rest/v1/feed';
```
Comment (mdziarkach) [inline src/repositories/meteorsRepository.js]: NASA_API_NEO_URL  created 2 times

**Review by mdziarkach — APPROVED** [review]

**Pull request merged**